oteapi_dlite/strategies/oceanlab_influx_parser.py

```diff
@@ -0,0 +1,279 @@
+from oteapi.models import AttrDict, HostlessAnyUrl, ParserConfig, ResourceConfig
+from pandas import DataFrame
+from pydantic import Field, SecretStr
+from pydantic.dataclasses import dataclass
+
+from oteapi_dlite.models import DLiteSessionUpdate
+from oteapi_dlite.utils import get_collection, update_collection
+from oteapi_dlite.utils.utils import get_meta
+
+if sys.version_info >= (3, 10):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
+
+
+class InfluxParseParseConfig(AttrDict):
+    """Configuration for DLite Excel parser."""
+
+    id: Annotated[
+        Optional[str], Field(description="Optional id on new instance.")
+    ] = None
+
+    label: Annotated[
+        Optional[str],
+        Field(
+            description="Optional label for new instance in collection.",
+        ),
+    ] = "json-data"
+
+    resourceType: Annotated[
+        Optional[Literal["resource/url"]],
+        Field(
+            description=ResourceConfig.model_fields["resourceType"].description,
+        ),
+    ] = "resource/url"
+    downloadUrl: Annotated[
+        Optional[HostlessAnyUrl],
+        Field(
+            description=ResourceConfig.model_fields["downloadUrl"].description,
+        ),
+    ] = None
+    mediaType: Annotated[
+        Optional[str],
+        Field(
+            description=ResourceConfig.model_fields["mediaType"].description,
+        ),
+    ] = None
+    storage_path: Annotated[
+        Optional[str],
+        Field(
+            description="Path to metadata storage",
+        ),
+    ] = None
+    collection_id: Annotated[
+        Optional[str], Field(description="A reference to a DLite collection.")
+    ] = None
+
+    url: Annotated[Optional[str], Field(description="url to the db")] = None
+
+    USER: Annotated[Optional[str], Field(description="user to the db")] = None
+
+    PASSWORD: Annotated[
+        Optional[SecretStr], Field(description="user pwd to the db")
+    ] = None
+
+    DATABASE: Annotated[Optional[str], Field(description="database name")] = (
+        None
+    )
+
+    RETPOLICY: Annotated[
+        Optional[str], Field(description="retpolicy for db")
+    ] = None
+
+    time_range: Annotated[
+        str, Field(description="timerange of values. eg : -12h")
+    ] = "-12h"
+
+    size_limit: Annotated[str, Field(description="rows to be extracted")] = "50"
+
+    measurements: Annotated[
+        list[dict],
+        Field(description="Measurement and field values as list of dictionary"),
+    ] = [
+        {
+            "measurement": "ctd_conductivity_munkholmen",
+            "field": "conductivity",
+        },
+        {
+            "measurement": "ctd_density_munkholmen",
+            "field": "density",
+        },
+        {
+            "measurement": "ctd_salinity_munkholmen",
+            "field": "salinity",
+        },
+        {
+            "measurement": "ctd_pressure_munkholmen",
+            "field": "pressure",
+        },
+    ]
```

Comment:
I still think making a separate data model for this would be better, no?

Something like

```python
class Measurement(BaseModel):
    """Measurement and field value from the Influx DB"""

    measurement: Annotated[str, Field(description="The measurement name")]
    field: Annotated[str, Field(description="The measurement's field in the DB")]
```

Then here it will be:
```suggestion
    measurements: Annotated[
        list[Measurement],
        Field(description="Measurement and field values as a list."),
    ] = [
        Measurement(_) for _ in [
            {
                "measurement": "ctd_conductivity_munkholmen",
                "field": "conductivity",
            },
            {
                "measurement": "ctd_density_munkholmen",
                "field": "density",
            },
            {
                "measurement": "ctd_salinity_munkholmen",
                "field": "salinity",
            },
            {
                "measurement": "ctd_pressure_munkholmen",
                "field": "pressure",
            },
        ]
    ]
```